appveyor.yml

```diff
@@ -59,8 +60,41 @@ test_script:
 - ps: |
     Write-Host "Tests are running..."
     Write-Host $pwd
-    & .\cibuild.cmd -test -integrationTest
-    if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
+    $testErrorsTotal = 0
+    $testRunCount = $env:TEST_RUN_COUNT
+    for ($testRun=1; $testRun -le $testRunCount; $testRun++)
+    {
+        Write-Host "[INFO]: Test Run ${testRun}/${testRunCount}, ${testErrorsTotal} test(s) failed before"
+
+        & .\cibuild.cmd -integrationTest
+        & .\cibuild.cmd -test
+
+        Get-ChildItem -recurse artifacts\bin\tests\TestResult.xml `
+        | ForEach-Object {
+            $resultFilePath = "$_"
+            Get-Content $resultFilePath `
+            | Select-String -Pattern '<test-run id="[^"]*"\s+testcasecount="(\d+)"\s+result="([^"]*)"\s+total="(\d+)"\s+passed="(\d+)"\s+failed="(\d+)".*' `
+            | foreach {
+                $result = $_.Matches.Groups
+                #$result[0].Value
+                $total  = $result[1].Value
+                $passed = $result[4].Value
+                $failed = $result[5].Value
+                $notPassed = $total - $passed
```

Comment:
This isn't used

Reply:
Yes, `$notPassed` would be a safe alternative to `$failed` as number of `$testErrors`. But it is too strict because we also have ignored / disabled testcases.

Reply:
...so remove?